File: VB/HowToSetRelationshipsBetweenObjects.Module/BusinessClasses.vb
Imports Microsoft.VisualBasic
Imports System

Imports DevExpress.Xpo

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation

Namespace HowToSetRelationshipsBetweenObjects.Module
	<DefaultClassOptions> _
	Public Class MasterObject
		Inherits XPObject
		Private name_Renamed As String
		Protected Overrides Sub OnSaving()
			MyBase.OnSaving()
		End Sub
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Public Property Name() As String
			Get
				Return name_Renamed
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", name_Renamed, value)
			End Set
		End Property
		<Association("A-A"), Aggregated> _
		Public ReadOnly Property AggregatedOneToManyCollection() As XPCollection(Of AggregatedOneToManyObject)
			Get
				Return GetCollection(Of AggregatedOneToManyObject)("AggregatedOneToManyCollection")
			End Get
		End Property
		<Association("B-B")> _
		Public ReadOnly Property NonAggregatedOneToManyCollection() As XPCollection(Of NonAggregatedOneToManyObject)
			Get
				Return GetCollection(Of NonAggregatedOneToManyObject)("NonAggregatedOneToManyCollection")
			End Get
		End Property
		<Association("C-C")> _
		Public ReadOnly Property NonAggregatedManyToManyCollection() As XPCollection(Of NonAggregatedManyToManyObject)
			Get
				Return GetCollection(Of NonAggregatedManyToManyObject)("NonAggregatedManyToManyCollection")
			End Get
		End Property
	End Class

	<DefaultClassOptions> _
	Public Class AggregatedOneToManyObject
		Inherits XPObject
		Private name_Renamed As String
		Private masterObject_Renamed As MasterObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Public Property Name() As String
			Get
				Return name_Renamed
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", name_Renamed, value)
			End Set
		End Property
		<Association("A-A")> _
		Public Property MasterObject() As MasterObject
			Get
				Return masterObject_Renamed
			End Get
			Set(ByVal value As MasterObject)
				SetPropertyValue("MasterObject", masterObject_Renamed, value)
			End Set
		End Property
	End Class

	<DefaultClassOptions> _
	Public Class NonAggregatedOneToManyObject
		Inherits XPObject
		Private name_Renamed As String
		Private masterObject_Renamed As MasterObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Public Property Name() As String
			Get
				Return name_Renamed
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", name_Renamed, value)
			End Set
		End Property
		<Association("B-B")> _
		Public Property MasterObject() As MasterObject
			Get
				Return masterObject_Renamed
			End Get
			Set(ByVal value As MasterObject)
				SetPropertyValue("MasterObject", masterObject_Renamed, value)
			End Set
		End Property
	End Class

	<DefaultClassOptions> _
	Public Class NonAggregatedManyToManyObject
		Inherits XPObject
		Private name_Renamed As String
		Protected Overrides Sub OnSaving()
			MyBase.OnSaving()
		End Sub
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Public Property Name() As String
			Get
				Return name_Renamed
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", name_Renamed, value)
			End Set
		End Property
		<Association("C-C")> _
		Public ReadOnly Property MasterObjects() As XPCollection(Of MasterObject)
			Get
				Return GetCollection(Of MasterObject)("MasterObjects")
			End Get
		End Property
	End Class
End Namespace

File: CS/HowToSetRelationshipsBetweenObjects.Module/BusinessClasses.cs
using System;

using DevExpress.Xpo;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;

namespace HowToSetRelationshipsBetweenObjects.Module {
	[DefaultClassOptions]
	public class MasterObject : XPObject {
		private String name;
		protected override void OnSaving() {
			base.OnSaving();
		}
		public MasterObject(Session session) : base(session) {
		}
		public String Name {
			get {
				return name;
			}
			set {
				SetPropertyValue("Name", ref name, value);
			}
		}
		[Association("A-A"), Aggregated]
		public XPCollection<AggregatedOneToManyObject> AggregatedOneToManyCollection {
			get {
				return GetCollection<AggregatedOneToManyObject>("AggregatedOneToManyCollection");
			}
		}
		[Association("B-B")]
		public XPCollection<NonAggregatedOneToManyObject> NonAggregatedOneToManyCollection {
			get {
				return GetCollection<NonAggregatedOneToManyObject>("NonAggregatedOneToManyCollection");
			}
		}
		[Association("C-C")]
		public XPCollection<NonAggregatedManyToManyObject> NonAggregatedManyToManyCollection {
			get {
				return GetCollection<NonAggregatedManyToManyObject>("NonAggregatedManyToManyCollection");
			}
		}
	}

	[DefaultClassOptions]
	public class AggregatedOneToManyObject : XPObject {
		private String name;
		private MasterObject masterObject;
		public AggregatedOneToManyObject(Session session) : base(session) {
		}
		public String Name {
			get {
				return name;
			}
			set {
				SetPropertyValue("Name", ref name, value);
			}
		}
		[Association("A-A")]
		public MasterObject MasterObject {
			get {
				return masterObject;
			}
			set {
				SetPropertyValue("MasterObject", ref masterObject, value);
			}
		}
	}

	[DefaultClassOptions]
	public class NonAggregatedOneToManyObject : XPObject {
		private String name;
		private MasterObject masterObject;
		public NonAggregatedOneToManyObject(Session session) : base(session) {
		}
		public String Name {
			get {
				return name;
			}
			set {
				SetPropertyValue("Name", ref name, value);
			}
		}
		[Association("B-B")]
		public MasterObject MasterObject {
			get {
				return masterObject;
			}
			set {
				SetPropertyValue("MasterObject", ref masterObject, value);
			}
		}
	}

	[DefaultClassOptions]
	public class NonAggregatedManyToManyObject : XPObject {
		private String name;
		protected override void OnSaving() {
			base.OnSaving();
		}
		public NonAggregatedManyToManyObject(Session session) : base(session) {
		}
		public String Name {
			get {
				return name;
			}
			set {
				SetPropertyValue("Name", ref name, value);
			}
		}
		[Association("C-C")]
		public XPCollection<MasterObject> MasterObjects {
			get {
				return GetCollection<MasterObject>("MasterObjects");
			}
		}
	}
}
